src/script/myRenderer.js

```diff
@@ -1,6 +1,12 @@
 import { Filter } from "./filter.js";
 
 export class Renderer {
+    
+    constructor (data) {
+        this.data = data;
+
+        const filter = new Filter(data);
```

Comment:
This line does nothing. It just creates an object and stores it in a variable until the function ends, which is on the next line, i.e. immediately.

If you did want to create a filter that can be used by the methods in this class (e.g. on line 77) I would recommend letting the user of this class create it and pass it into the constructor. That's a pattern called "dependency injection", which means that if this class needs ("depends on") something, it's injected by the code that instantiates this class, meaning that this class doesn't need to know exactly what the type of object is, only what _role_ it plays (i.e. what methods it has, etc).